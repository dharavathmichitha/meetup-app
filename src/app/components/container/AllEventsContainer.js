import React from 'react';
import PropTypes from 'prop-types';
import Events from '../presentational/events/Events';
import { useEventsListData, useEventsCities } from '../../hooks/event-hooks';
import EventFilters from '../presentational/events/EventFilters';

export default function AllEventsContainer() {
  const [eventsListData] = useEventsListData();
  const [cityDataList] = useEventsCities();
  return (
    <>
      <EventFilters />
      <Events cssClass="" cityDataList={cityDataList || []} eventsListData={eventsListData || []} />
    </>
  );
}

AllEventsContainer.prototype = {
  eventsListData: PropTypes.array.isRequired,
};
