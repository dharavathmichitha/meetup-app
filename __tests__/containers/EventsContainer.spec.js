import React from 'react';
import TestRenderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import axios from 'axios';
import AllEventsContainer from '../../src/app/components/container/AllEventsContainer';

beforeEach(() => {
  jest.resetModules();
});

afterEach(cleanup);
describe('AllEventsContainer', () => {
  it('should render container without events included', done => {
    const mockSuccessResponse = {
      events: [
        {
          id: 0,
          isFree: true,
          name: 'CSS Grids: fact or fiction',
          city: 9,
          startDate: '2019-07-14T02:00:00+00:00',
          endDate: '2019-07-14T03:00:00+00:00',
        },
      ],
    };
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
    });
    jest.spyOn(axios, 'get').mockImplementation(() => mockFetchPromise);

    const container = TestRenderer.create(<AllEventsContainer />);

    expect(axios.get).toHaveBeenCalledTimes(0);

    const containerTree = container.toJSON();

    expect(containerTree).toMatchSnapshot();
    done();
  });
});
