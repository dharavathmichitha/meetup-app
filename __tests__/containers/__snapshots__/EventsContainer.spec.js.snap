// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllEventsContainer should render container without events included 1`] = `
Array [
  <section
    className="filters"
  >
    <input
      aria-label="Close"
      className="filters__event-search"
      id="searchinput"
      onChange={[Function]}
      placeholder="Search an event..."
      type="text"
    />
    <button
      className="filters__free-events btn btn"
      data-hover="Cancel"
      onClick={[Function]}
      type="button"
    >
      Discover Free Events
    </button>
  </section>,
  <section
    className="events "
  />,
]
`;
